src/java/org/apache/cassandra/io/util/MmappedRegions.java

```diff
@@ -152,15 +152,15 @@ private void updateState(long length)
         }
     }
 
-    private void updateState(CompressionMetadata metadata)
+    private void updateState(CompressionMetadata metadata, long startOffset)
     {
         long offset = 0;
-        long lastSegmentOffset = 0;
+        long lastSegmentOffset = state.startOffset;
         long segmentSize = 0;
 
         while (offset < metadata.dataLength)
         {
-            CompressionMetadata.Chunk chunk = metadata.chunkFor(offset + metadata.uncompressedOffset);
+            CompressionMetadata.Chunk chunk = metadata.chunkFor(offset + startOffset);
```

Comment:
Could we name the two `startOffset`s differently? E.g. `compressedStartOffset` vs `uncompressedStartOffset`?

Also, `offset` and `lastSegmentOffset` are confusingly similarly named, while they are in different spaces. Could they become e.g. `uncompressedPosition` and `compressedSegmentStartPosition`?